entity.go

```diff
@@ -77,6 +77,26 @@ func NewMultipart(header Header, parts []*Entity) (*Entity, error) {
 	return New(header, r)
 }
 
+// ReadOptions represents the options in a message reader
+type ReadOptions struct {
+	Header textproto.ReadOptions
+}
+
+// ReadWithOptions is a mirror of Read to read a message from r with options.
+//
+// Default value for header option of MaxHeaderLineLength
```

Comment:
I don't think documenting a specific `ReadOptions` field is necessary here. Users can just click on `ReadOptions` in GoDoc to discover available options.